sklearn/metrics/ranking.py

```diff
@@ -292,7 +299,23 @@ def _binary_roc_auc_score(y_true, y_score, sample_weight=None):
 
         fpr, tpr, tresholds = roc_curve(y_true, y_score,
                                         sample_weight=sample_weight)
-        return auc(fpr, tpr)
+        if max_fpr is None or max_fpr == 1:
+            return auc(fpr, tpr)
+
+        idx = np.where(fpr <= max_fpr)[0]
+        # Linearly interpolate the ROC curve until max_fpr
+        idx_in = idx[-1]
+        idx_out = idx_in + 1
+        x_interp = [fpr[idx_in], fpr[idx_out]]
+        y_interp = [tpr[idx_in], tpr[idx_out]]
+        tpr = np.append(tpr[idx], np.interp(max_fpr, x_interp, y_interp))
+        fpr = np.append(fpr[idx], max_fpr)
+        partial_auc = auc(fpr, tpr)
+
+        # McClish correction: standardize result to lie between 0.5 and 1
```

Comment:
It seems that the computation of `mean_area` assumes that the estimator is always better than random guess. In practice, for a very bad estimator (doing worst than random guess) `partial_auc` could be less than `min_area` and the result will not necessarily be in [0.5, 1]. It will always be lower than 1 but can be less than 0.5 which would mean that we are doing worse than a random guess estimator. I think we should keep this standardization but we should explain somewhere that the result is not necessarily in [0.5, 1].